File: python-examples/browser-sdk-showcase/api/wait-for-network-settled.py
# https://docs.intunedhq.com/automation-sdks/intuned-sdk/python/helpers/functions/wait_for_network_settled
from typing import TypedDict

from intuned_browser import wait_for_network_settled
from playwright.async_api import Page


# Decorator without arguments (uses timeout_s=30, max_inflight_requests=0)
@wait_for_network_settled
async def click_load_more(page):
    await page.locator("main main button").click()


class Params(TypedDict):
    pass


async def automation(page: Page, params: Params, **_kwargs):
    await page.goto("https://sandbox.intuned.dev/load-more")
    # Automatically waits for network to settle after clicking
    await click_load_more(page)
    # Network has settled, data is loaded

File: typescript-examples/browser-sdk-showcase/api/wait-for-network-settled.ts
// https://docs.intunedhq.com/automation-sdks/intuned-sdk/typescript/helpers/functions/withNetworkSettledWait
import { withNetworkSettledWait } from "@intuned/browser";
import { BrowserContext, Page } from "playwright";

interface Params {}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  await page.goto("https://sandbox.intuned.dev/infinite-scroll");

  // Execute action and wait for network to settle
  const result = await withNetworkSettledWait(
    async (page) => {
      // scroll to load more content
      await page.evaluate(() => {
        window.scrollTo(0, document.body.scrollHeight);
      });
      return "scrolled";
    },
    {
      page,
      timeoutInMs: 15000,
      maxInflightRequests: 0,
    }
  );
  console.log(result); // "scrolled"
  return result;
}